tests/ecn/files/helper.py

```diff
@@ -0,0 +1,345 @@
+
+    """
+
+    result = list()
+
+    rx_port_id = port_id
+    tx_port_id = (port_id + 1) % len(testbed_config.devices)
+
+    data_endpoint = DeviceTxRx(
+        tx_device_names=[testbed_config.devices[tx_port_id].name],
+        rx_device_names=[testbed_config.devices[rx_port_id].name],
+    )
+
+    data_flow_delay_nanosec = sec_to_nanosec(data_flow_delay_sec)
+
+    """ Data Flow """
+    ip_prio = Priority(Dscp(phb=FieldPattern(choice=prio_dscp_map[prio]),
+                            ecn=FieldPattern(choice=Dscp.ECN_CAPABLE_TRANSPORT_1)))
+    pfc_queue = FieldPattern([prio])
+
+    data_flow = Flow(
+        name=data_flow_name,
+        tx_rx=TxRx(data_endpoint),
+        packet=[
+            Header(choice=EthernetHeader(pfc_queue=pfc_queue)),
+            Header(choice=Ipv4Header(priority=ip_prio))
+        ],
+        size=Size(data_pkt_size),
+        rate=Rate('line', 100),
+        duration=Duration(FixedPackets(packets=data_pkt_cnt,
+                                       delay=data_flow_delay_nanosec,
+                                       delay_unit='nanoseconds'))
+    )
+
+    result.append(data_flow)
+
+    """ PFC Pause Storm """
+    pause_time = []
+    for x in range(8):
+        if x == prio:
+            pause_time.append('ffff')
+        else:
+            pause_time.append('0000')
+
+    vector = pfc_class_enable_vector([prio])
+    pause_pkt = Header(PfcPause(
+        dst=FieldPattern(choice='01:80:C2:00:00:01'),
+        src=FieldPattern(choice='00:00:fa:ce:fa:ce'),
+        class_enable_vector=FieldPattern(choice=vector),
+        pause_class_0=FieldPattern(choice=pause_time[0]),
+        pause_class_1=FieldPattern(choice=pause_time[1]),
+        pause_class_2=FieldPattern(choice=pause_time[2]),
+        pause_class_3=FieldPattern(choice=pause_time[3]),
+        pause_class_4=FieldPattern(choice=pause_time[4]),
+        pause_class_5=FieldPattern(choice=pause_time[5]),
+        pause_class_6=FieldPattern(choice=pause_time[6]),
+        pause_class_7=FieldPattern(choice=pause_time[7]),
+    ))
+
+    """ Pause frames are sent from the RX port """
+    pause_src_point = PortTxRx(tx_port_name=testbed_config.ports[rx_port_id].name,
+                               rx_port_name=testbed_config.ports[tx_port_id].name)
+
+    speed_str = testbed_config.layer1[0].speed
+    speed_gbps = int(speed_str.split('_')[1])
+    pause_dur = 65535 * 64 * 8.0 / (speed_gbps * 1e9)
+    pps = int(2 / pause_dur)
+
+    pause_flow = Flow(
+        name=pause_flow_name,
+        tx_rx=TxRx(pause_src_point),
+        packet=[pause_pkt],
+        size=Size(64),
+        rate=Rate('pps', value=pps),
+        duration=Duration(FixedSeconds(seconds=exp_dur_sec,
+                                       delay=0,
+                                       delay_unit='nanoseconds'))
+    )
+
+    result.append(pause_flow)
+    return result
+
+
+def __config_capture(testbed_config, port_id):
```

Comment:
rename to specify that capture is only for ip pkts

Reply:
Fixed